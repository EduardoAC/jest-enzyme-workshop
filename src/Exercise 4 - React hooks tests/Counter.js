import React from "react";

export const Counter = () => {
  const [counter, setCounter] = React.useState(0);
  const increase = React.useCallback(() => {
    setCounter(counter + 1);
  }, [counter, setCounter]);
  const decrease = React.useCallback(() => {
    setCounter(counter - 1);
  }, [counter, setCounter]);

  return (
    <div>
      <button onClick={decrease}>-</button>
      <span>{` ${counter} `}</span>
      <button onClick={increase}>+</button>
    </div>
  );
};
