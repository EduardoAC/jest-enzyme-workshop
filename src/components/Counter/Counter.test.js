import React from 'react'
import { shallow } from 'enzyme'
import { Counter } from './Counter'

/**
 * What can we do to make the test more readable?
 */
describe('<Counter />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Counter />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should increase counter by clicking + button', () => {
    const wrapper = shallow(<Counter />)
    wrapper.find('button').last().simulate('click')
    expect(wrapper.find('span').text()).toBe(' 1 ')
  })
  it('should decreate counter by clicking - button', () => {
    const wrapper = shallow(<Counter />)
    wrapper.find('button').first().simulate('click')
    expect(wrapper.find('span').text()).toBe(' -1 ')
  })
})