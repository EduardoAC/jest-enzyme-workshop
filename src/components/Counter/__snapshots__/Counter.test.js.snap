// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should match snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <span>
     0 
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
